tests/cl-go-client/main.go

```diff
@@ -82,26 +82,33 @@ func main() {
 
 	rand.Seed(randSeed)
 
+	var wg sync.WaitGroup
 	for i := 0; i < numPositions; i++ {
-		var (
-			// 1 to 9. These are localosmosis keyring test accounts with names such as:
-			// lo-test1
-			// lo-test2
-			// ...
-			randAccountNum = rand.Intn(8) + 1
-			accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
-			// minTick <= lowerTick <= upperTick
-			lowerTick = rand.Int63n(maxTick-minTick+1) + minTick
-			// lowerTick <= upperTick <= maxTick
-			upperTick = maxTick - rand.Int63n(int64(math.Abs(float64(maxTick-lowerTick))))
-
-			tokenDesired0 = sdk.NewCoin(denom0, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
-			tokenDesired1 = sdk.NewCoin(denom1, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
-		)
-
-		log.Println("creating position: pool id", expectedPoolId, "accountName", accountName, "lowerTick", lowerTick, "upperTick", upperTick, "token0Desired", tokenDesired0, "tokenDesired1", tokenDesired1, "defaultMinAmount", defaultMinAmount)
-		amt0, amt1, liquidity := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokenDesired0, tokenDesired1, defaultMinAmount, defaultMinAmount)
-		log.Println("created position: amt0", amt0, "amt1", amt1, "liquidity", liquidity)
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			var (
+				// 1 to 9. These are localosmosis keyring test accounts with names such as:
+				// lo-test1
+				// lo-test2
+				// ...
+				randAccountNum = rand.Intn(8) + 1
+				accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
+				// minTick <= lowerTick <= upperTick
+				lowerTick = rand.Int63n(maxTick-minTick+1) + minTick
+				// lowerTick <= upperTick <= maxTick
+				upperTick = maxTick - rand.Int63n(int64(math.Abs(float64(maxTick-lowerTick))))
+
+				tokenDesired0 = sdk.NewCoin(denom0, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
+				tokenDesired1 = sdk.NewCoin(denom1, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
+			)
+
+			log.Println("creating position: pool id", expectedPoolId, "accountName", accountName, "lowerTick", lowerTick, "upperTick", upperTick, "token0Desired", tokenDesired0, "tokenDesired1", tokenDesired1, "defaultMinAmount", defaultMinAmount)
+			amt0, amt1, liquidity := createPosition(igniteClient, expectedPoolId, accountName, lowerTick, upperTick, tokenDesired0, tokenDesired1, defaultMinAmount, defaultMinAmount)
+			log.Println("created position: amt0", amt0, "amt1", amt1, "liquidity", liquidity)
+		}(i)
+
+		wg.Wait()
```

Comment:
I think `wg.Wait` should be outside of the loop for the concurrency benefits to become apparent. Otherwise, we are blocking every iteration here

Reply:
Yeah actually it should be outside the loop, thanks for the review, fixed in the most recent commit